spec/requests/organization_requests_spec.rb

```diff
@@ -0,0 +1,171 @@
+require "rails_helper"
+
+RSpec.describe "Organizations", type: :request do
+  let(:default_params) do
+    { organization_id: @organization.to_param }
+  end
+
+  context "While signed in as a normal user" do
+    before do
+      sign_in(@user)
+    end
+
+    describe "GET #show" do
+      before { get organization_path(default_params) }
+
+      it { expect(response).to be_successful }
+
+      it 'load the current organization' do
+        expect(assigns(:organization)).to have_attributes(
+          name: @organization.name,
+          email: @organization.email,
+          url: @organization.url
+        )
+      end
+    end
+
+    describe "GET #edit" do
+      before { get edit_organization_path(default_params) }
+
+      it { expect(response).to have_http_status(:redirect) }
+      it { expect(response).to have_error }
```

Comment:
Is there a way to check which error is here?

If yes, lets try to add it to the test

Reply:
I was looking because I thought there was a way to check for specific error codes but my google foo is failing me.

Reply:
I've tried to check the message or the exception, but it isn't possible with `have_error`. And when I try to use the matcher `raise_error` the error does not exist:

```rb
Failures:

  1) Organizations While signed in as a normal user GET #edit is expected to raise Exception with "AccessDenied"
     Failure/Error: it { expect { response }.to raise_error('AccessDenied') }
       expected Exception with "AccessDenied" but nothing was raised
```

I guess the error is about the HTTP status 302 - redirect. So it isn't an exception or similar.

Reply:
I actually think it's fine to utilize the `have_http_status` to determine if authorization worked or not. In some perspectives, you could say this wasn't "an error" we intend for this to happen.